packages/components/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react"
+import { ScaledSize, ViewStyle, useWindowDimensions } from "react-native";
+import { EDSStyleSheet } from "../../styling";
+import { useStyles } from "../../hooks/useStyles";
+import { Paper } from "../Paper";
+import { Scrim, ScrimProps } from "../Scrim";
+
+type DialogProps = Omit<ScrimProps, "onPress"> & { style?: ViewStyle } & { onScrimPress?: ScrimProps["onPress"] }
+
+export const Dialog = ({ isOpen, style, onScrimPress, children }: DialogProps) => {
+    const dimensions = useWindowDimensions();
+    const styles = useStyles(themeStyles, { style, dimensions })
+    return <Scrim isOpen={isOpen} onPress={onScrimPress}>
+        <Paper elevation="aboveScrim" style={styles.dialog}>
```

Comment:
Styles are one common prop that users might find themselves trying to overwrite when using the component, but there are actually a lot of other props defined on `View` that could be handy like accessibility and touch event behaviour configuration. In essence, the Dialog component is a View wrapper since it basically consists of a `Paper` component (which is most definately a styled `View` wrapper). Because of this, I think we should define the props for the Dialog component like this:

```tsx
// Note the missing style prop:
type DialogProps = Omit<ScrimProps, "onPress"> & {onScrimPress?: ScrimProps["onPress"]}; 

// Notice the typing for the props now:
const Dialog = ({
  isOpen,
  onScrimPress,
  children,
  ...rest,
}: DialogProps & ViewProps) => {
  // .. 
  return (
    <Scrim isOpen{isOpen} onPress{onScrimPress}>
    // Note how we spread out the remaining ViewProps into the Paper component (which acceps ViewProps)
      <Paper elevation="aboveScrim" style={[styles.dialog, rest.styles]} {...rest}>
        {children}
      </Paper>
    </Scrim>
  )
}
```